### PR TITLE
Shopify best practices

### PR DESCRIPTION
**Pull request by ishachopra0401**
## Description of the change

Shopify best practices

## Type of change
- [ ] Bug fix (non-breaking change that fixes an issue)
- [ ] New feature (non-breaking change that adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Related issues

> Fix [#1]() 

### Development

- [ ] Lint rules pass locally
- [ ] The code changed/added as part of this pull request has been covered with tests
- [ ] All tests related to the changed code pass in development

### Code review 

- [ ]  This pull request has a descriptive title and information useful to a reviewer. There may be a screenshot or screencast attached
- [ ] "Ready for review" label attached to the PR and reviewers mentioned in a comment
- [ ] Changes have been reviewed by at least one other engineer
- [ ] Issue from task tracker has a link to this pull request

### CODE REVIEW

**Comment by yashasvibajpai** [comment]
Adding @utsabc for addtional reviewing here,

@utsabc WDYT on best practices as a section in the Shopify destination doc rather than a separate linked document?

**Comment by utsabc** [comment]
> Adding @utsabc for addtional reviewing here,
> 
> @utsabc WDYT on best practices as a section in the Shopify destination doc rather than a separate linked document?

I think its fine for now its a short writeup, which can be consolidated in a single page

**Review by utsabc — APPROVED** [review]

**Review by yashasvibajpai — APPROVED** [review]

File: docs/stream-sources/shopify.mdx [hunk 1]
```diff
@@ -313,6 +319,29 @@ The below is an example of client-side event transformed by RudderStack:
 }
 ```
 
+
+## Best practices
+
+### Provide login functionality
+
+Shopify provides store owners with the capability to either make user login compulsory or allow guest checkout in their stores. However, it is a best practice to prompt  users to login on your platform. That ways, RudderStack fetches the user details to provide you with meaningful event data corresponding to the user's journey on your platform.
```
Comment (ameypv-rudder) [inline docs/stream-sources/shopify.mdx:327]:
```suggestion
Shopify provides the store owners the capability to either make user login compulsory or allow guest checkout in their stores. However, it is a best practice to prompt users to log in onto your platform. That way, RudderStack can fetch the user details and provide you with meaningful event data corresponding to the user's journey on your platform.
```

File: docs/stream-sources/shopify.mdx [hunk 2]
```diff
@@ -313,6 +319,29 @@ The below is an example of client-side event transformed by RudderStack:
...
+
+### Support for client side calls
```
Comment (ameypv-rudder) [inline docs/stream-sources/shopify.mdx:329]:
```suggestion
### Support for client-side calls
```

File: docs/stream-sources/shopify.mdx [hunk 3]
```diff
@@ -313,6 +319,29 @@ The below is an example of client-side event transformed by RudderStack:
...
+
+It is recommended **NOT** to make any drastic changes to your store structure in order to preserve data integrity. If done, RudderStack may lead to incorrect or missing calls owing to the disparity between URL and triggered event. 
+
+Refer to the following additional details for e-commerce events:
```
Comment (ameypv-rudder) [inline docs/stream-sources/shopify.mdx:333]:
```suggestion
Note to the following additional details for e-commerce events tracked by RudderStack:
```

File: docs/stream-sources/shopify.mdx [hunk 4]
```diff
@@ -337,6 +366,48 @@ You can try deleting and reinstalling the app. If that does not work, please [co
 
 You need to have the app installed and configured properly in each of the stores.
 
+### What is the difference between `browser_ip` and `requestIP`?
+
+The `requestIP` is the IP address of Shopify backend server which pushes data to RudderStack while the `browser_ip` is the IP address of user's browser.
```
Comment (ameypv-rudder) [inline docs/stream-sources/shopify.mdx:371]:
```suggestion
The `requestIP` is the IP address of the Shopify backend server which pushes data to RudderStack, while `browser_ip` is the IP address of the user.
```

File: docs/stream-sources/shopify.mdx [hunk 5]
```diff
@@ -337,6 +366,48 @@ You can try deleting and reinstalling the app. If that does not work, please [co
...
+
+### Is it possible to collect additional/custom fields from Shopify?
+
+Yes, it is possible to collect additional/custom fields from Shopify by using `identify` and `track` calls.
+
+Note that the `analytics` object of JS SDK (for example, `rudderanalytics.track()` or `rudderanalytics.identify()`) is already loaded on the `window` object of the store where the app is installed. Hence, the `analytics` object should not be loaded separately to collect additional/custom fields from Shopify.
+
+### How does RudderStack set Anonymous ID and can it persist?
```
Comment (ameypv-rudder) [inline docs/stream-sources/shopify.mdx:379]:
```suggestion
### How does RudderStack set the `anonymousId`? Does RudderStack persist it for future calls?
```

File: docs/stream-sources/shopify.mdx [hunk 6]
```diff
@@ -337,6 +366,48 @@ You can try deleting and reinstalling the app. If that does not work, please [co
...
+
+RudderStack fetches Shopify's `customerId` and uses it as `anonymousId` to allow identity stitching in downstream destinations.
```
Comment (ameypv-rudder) [inline docs/stream-sources/shopify.mdx:381]:
```suggestion
RudderStack fetches Shopify's `customerId` and uses it as `anonymousId` to allow identity stitching in the downstream destinations.
```

File: docs/stream-sources/shopify.mdx [hunk 7]
```diff
@@ -337,6 +366,48 @@ You can try deleting and reinstalling the app. If that does not work, please [co
...
+
+No, the `anonymousId` does not persist and keeps on changing as the Shopify servers push data to RudderStack over server to server API's.
```
Comment (ameypv-rudder) [inline docs/stream-sources/shopify.mdx:383]:
```suggestion
No, RudderStack does not persist the `anonymousId` - it keeps on changing as the Shopify servers push data to RudderStack over the server-to-server APIs.
```

File: docs/stream-sources/shopify.mdx [hunk 8]
```diff
@@ -337,6 +366,48 @@ You can try deleting and reinstalling the app. If that does not work, please [co
...
 
+### What is the difference between `browser_ip` and `requestIP`?
+
+The `requestIP` is the IP address of Shopify backend server which pushes data to RudderStack while the `browser_ip` is the IP address of user's browser.
+
+### Is it possible to collect additional/custom fields from Shopify?
+
+Yes, it is possible to collect additional/custom fields from Shopify by using `identify` and `track` calls.
+
+Note that the `analytics` object of JS SDK (for example, `rudderanalytics.track()` or `rudderanalytics.identify()`) is already loaded on the `window` object of the store where the app is installed. Hence, the `analytics` object should not be loaded separately to collect additional/custom fields from Shopify.
```
Comment (ameypv-rudder) [inline docs/stream-sources/shopify.mdx:377]:
```suggestion
Note that the `rudderanalytics` object of JS SDK (for example, `rudderanalytics.track()` or `rudderanalytics.identify()`) is already loaded on the `window` object of the store where the app is installed. Hence, the `rudderanalytics` object should not be loaded separately to collect additional/custom fields from Shopify.
```

Comment (ameypv-rudder) [reply docs/stream-sources/shopify.mdx:377]: Confirmed with @yashasvibajpai re: this change. cc @utsabc @ishachopra0401

File: docs/stream-sources/shopify.mdx [hunk 9]
```diff
@@ -337,6 +366,48 @@ You can try deleting and reinstalling the app. If that does not work, please [co
...
+
+### How does RudderStack set Anonymous ID and can it persist?
+
+RudderStack fetches Shopify's `customerId` and uses it as `anonymousId` to allow identity stitching in downstream destinations.
+
+No, the `anonymousId` does not persist and keeps on changing as the Shopify servers push data to RudderStack over server to server API's.
+
+### How can I differentiate between data collected from Shopify Cloud (Webhooks) and Shopify Device (Web Device Mode)?
```
Comment (ameypv-rudder) [inline docs/stream-sources/shopify.mdx:385]:
```suggestion
### How can I differentiate between the data collected from Shopify Cloud (Webhooks) and Shopify Device (web device mode)?
```

File: docs/stream-sources/shopify.mdx [hunk 10]
```diff
@@ -337,6 +366,48 @@ You can try deleting and reinstalling the app. If that does not work, please [co
...
+
+You can differentiate between the data collected in cloud and device mode by looking at the signature in payload:
+
+- In cloud mode, RudderStack subscribes to Shopify webhooks to consume data from Shopify. Data collected from Shopify web has the following signature in the payload:
```
[the remark this answers is not in the document]
Comment (ameypv-rudder) [reply docs/stream-sources/shopify.mdx:389]: @ishachopra0401  In the last sentence, should we change "Shopify web" to "Shopify Cloud"? To be consistent with the term used in the FAQ and in the payload below.

**Review by ameypv-rudder — APPROVED** [review]

**Pull request merged**